src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala

```diff
@@ -89,14 +89,19 @@ class DepositServlet(app: EasyDepositApiApp) extends ProtectedServlet(app) {
   }
   post("/:uuid/file/*") { //file
     {
-      for { // a unique request signature so no forXxx method
+      val zip = "application/zip"
+      val bin = "application/octet-stream"
+      val msg = s"expecting header 'Content-Type: $zip' or 'Content-Type: $bin'; the latter with a 'Content-Disposition'."
+      for {
         managedIS <- getRequestBodyAsManagedInputStream
-        uuid <- getUUID
-        dir <- getPath
-        // TODO EASY-1658: mime type application/zip or application/binary
-        disposition <- getMandatoryHeader("Content-Disposition")
-        path = dir.resolve(disposition)
-        newFileWasCreated <- managedIS.apply(is => app.writeDepositFile(is)(user.id, uuid, path))
+        maybeContentDisposition = getLowerCaseHeaderValue("Content-Disposition")
+        maybeContentType = getLowerCaseHeaderValue("Content-Type")
+        newFileWasCreated <- (maybeContentType, maybeContentDisposition) match {
+          case (Some(`zip`), _) => Failure(???) // TODO issue EASY-1658
+          case (Some(`bin`), Some(contentDisposition: String)) =>
+            managedIS.apply(is => forPathSupplement(contentDisposition, app.writeDepositFile(is)))
```

Comment:
this, as well as the implementation of `forPathSupplement`, looks more complicated than it should probably be.

1. I don't think you require a callback here
2. you're opening the `managedIS` using `apply` way too early. After all, `forPathSupplement` might already fail in the first 3 parts for the for-comprehension.
3. In the implementation of `forPathSupplement` I don't understand why you wrap the callback call in a `Try` and flatten it immediately. The comment there isn't clear either.

Reply:
I tried to replace
https://github.com/DANS-KNAW/easy-deposit-api/blob/eb783743d1ac3359f394477c4cf26b2ce25acf96/src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala#L34
https://github.com/DANS-KNAW/easy-deposit-api/blob/eb783743d1ac3359f394477c4cf26b2ce25acf96/src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala#L39
with 
```
app.getDeposits(user.id)
app.createDeposit(user.id)
```
that broke a unit test with an unexpected exception because somewhere something was not 'honest'. So I decided to apply the same old pattern here. We might need to rethink that pattern.